lib/auth/keystore/gcp_kms_test.go

```diff
@@ -0,0 +1,367 @@
+// Copyright 2022 Gravitational, Inc
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package keystore
+
+import (
+	"context"
+	"crypto"
+	"crypto/rand"
+	"errors"
+	"sync"
+	"testing"
+
+	gax "github.com/googleapis/gax-go/v2"
+	"github.com/jonboulle/clockwork"
+	"github.com/stretchr/testify/require"
+	"google.golang.org/api/iterator"
+	kmspb "google.golang.org/genproto/googleapis/cloud/kms/v1"
+)
+
+type mockGCPKMSClient struct {
+	// map of key version id to state
+	keyVersions             map[string]string
+	clock                   clockwork.FakeClock
+	activateKey             chan struct{}
+	keyActivated            chan struct{}
+	unblockGetPublicKey     chan struct{}
+	unblockDestroyCryptoKey chan struct{}
+	unblockIterator         chan struct{}
+	signIsBlocked           chan struct{}
+	keyIsDestroyed          chan struct{}
+	wg                      sync.WaitGroup
+	sync.RWMutex
```

Comment:
nit: needless embedding? It's ok for a test, but in general we shouldn't embed if we don't plan to inherit/expose the exact same methods the embed carries. In this case, it looks me that the lock is an implementation detail and not something a user of mockGCPKMSClient should mess with.